### ISSUE TITLE
Issue with FCC "Bonfire: Make a Person" & Camperbot solutions

### ISSUE THREAD

**Issue by soulchainer** [FreeCodeCamp/FreeCodeCamp#3370]
Hi. I think both solutions are wrong. Because Camperbot solution is wrong and FCC bonfire accepts camperbot solution.

BTW: I didn't want to watch the definitive solution, just a hint, but the FCC link derived me to the channel and that was the first thing I saw, my only mistake was being using directly the parameter instead of assign it previously to a variable.

Camperbot says that this is a solution:

```js
var Person = function(firstAndLast) {

    var fullName = firstAndLast;
    var arr = fullName.split(' ');

    this.getFirstName = function() {
        return arr[0];
    };
    this.getLastName = function() {
        return arr[1];
    };
    this.getFullName = function() {
        return fullName;
    };
    this.setFirstName = function(first) {
        arr[0] = first;
    };
    this.setLastName = function(last) {
        arr[1] = last;
    };
    this.setFullName = function(firstAndLast) {
        fullName = firstAndLast;
    };
};
```

Well, it sure pass all the tests of FCC, because they are done for passing that solution, but in this solution **partial setters doesn't set**. Oh, yes: them set the *firstname* and *lastname*, for sure.
But if you change the *firstname* or the *lastname*, you *should also update the fullname*.
So, like it don't do it, you «set» the *firstname* and check after that the *fullname* of your person and you get that, well, *nothing has changed*.

When you set the firstname of lastname individually, you should also update the fullname to reflect these changes. This would make it:

```js
this.setFirstName = function(first) {
    arr[0] = first;
    this.setFullName(names.join(" "));
  };
  this.setLastName = function(last) {
    arr[1] = last;
    this.setFullName(names.join(" "));
  };
```

Funny thing is that this also validates, so perfect xD. But I think you should fix the camperbot hint and fix the tests, for check for fullname update also.

**Note:** I'm not saying it's your *duty* to doing it waving my fist dangerously. If you agree I'm correct in this and nobody is going to race me, I could fix it myself tomorrow, I guess. But I should read about how to make the extra tests, because I don't know anything about making tests.

**Comment by natac13** [comment]
I believe this would also coinside with what you have said
```js
var Person = function(firstAndLast) {
  var name_arr = firstAndLast.split(' ');
  var first_name = name_arr[0];
  var last_name = name_arr[1];
  this.setFirstName = function(value) { first_name = value; };
  this.setLastName = function(value) { last_name = value; };
  this.setFullName = function(value) {
    var arr = value.split(' ');
    first_name = arr[0];
    last_name = arr[1];
  };
  this.getFirstName = function() { return first_name; };
  this.getLastName = function() { return last_name; };
  this.getFullName = function() { return first_name + " " + last_name; };
};
```

since the first and last name can be changed from the set function and when I call for `getFullName` it combines them at that time. 

I do agree that the solution that camperbot gives does not reflect any changes made to the first or last name when looking at the full name.

**Comment by soulchainer** [comment]
@natac13 Well... not. Because you don't
> since the first and last name can be changed from the set function and when I call for getFullName it combines them at that time.

You don't do that. You only simulates you do that. And in a wrong way, because getters are for **only** show values, not creating them. You should change the values with the setters. That's the function of the setters.

**Comment by natac13** [comment]
Hmm ok.

**Comment by tarzak** [comment]
+1 solution in the wiki is wrong. Please fix the tests.

**Comment by soulchainer** [comment]
@benschenker This isn't only a wiki issue, but also a tests issue.

**Comment by ltegman** [comment]
I've put in a pull request to fix these tests and have updated the solution on the wiki.

**Comment by ltegman** [comment]
Closing this issue as I just merged PR #3562 to solve this problem. Let me know if you run into any other issues with this bonfire.

Thanks and happy coding!
